src/createServer.js

```diff
@@ -1,9 +1,43 @@
 /* eslint-disable no-console */
 'use strict';
 
+const http = require('http');
+
 function createServer() {
   /* Write your code here */
   // Return instance of http.Server class
+
+  const server = http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+    res.statusCode = 200;
+
+    res.on('error', (error) => {
+      console.error(error);
+      res.statusCode = 400;
+      res.end();
+    });
+
+    const url = new URL(req.url.slice(1), `http://${req.headers.host}`);
```

Comment:
The URL parsing assumes that the request URL is always correctly formatted. Consider adding a try-catch block to handle potential errors when parsing the URL to prevent the server from crashing if the URL is malformed.